processor/transformprocessor/config.go

```diff
@@ -0,0 +1,27 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transformprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor"
+
+import (
+	"go.opentelemetry.io/collector/config"
+)
+
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"`
+
+	Queries []string `mapstructure:"queries"`
+}
```

Comment:
Should we have a "Validate" maybe to ensure the query are parsable? 

You can make this `Queries []Query` and make Query implement https://pkg.go.dev/encoding#TextUnmarshaler, the config unmarshaler will do the right thing.

See https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/identifiable.go#L73 or https://github.com/open-telemetry/opentelemetry-collector/blob/51fcb65f02055638f223c6abd3d6d2d644c696db/config/configtelemetry/configtelemetry.go#L96

Reply:
We are looking to implement a "--dry-run" and that would not be possible if you can return error only at "start" time.